java/daikon/chicory/Runtime.java

```diff
@@ -439,7 +446,12 @@ public static void noMoreOutput() {
     // The incrementRecords method (which calls this) is called inside a
     // synchronized block, but re-synchronize just to be sure, or in case
     // this is called from elsewhere.
-    synchronized (Runtime.dtrace) {
+
+    // Ensure that the following method calls on dtrace all occur on the same
```

Comment:
Here and two other places with a similar comment, the purpose if this code is not to ensure this property; we already believe the property to be true.  The code is a workaround because typechecker doesn't know that the side-effecting methods don't reset Runtime.dtrace.  If the field were reassigned, then the use of the local variable would mask the error rather than be useful.  Please update the comment.